src/message/messageActionSheet.js

```diff
@@ -226,31 +225,38 @@ const cancel = params => {};
 cancel.title = 'Cancel';
 cancel.errorMessage = 'Failed to hide menu';
 
-export const constructHeaderActionButtons = ({
-  backgroundData: { mute, subscriptions, ownUser },
-  stream,
+export const constructTopicActionButtons = ({
+  backgroundData: { mute, subscriptions, ownUser, unreadStreams },
   topic,
+  streamId,
 }: {|
   backgroundData: $ReadOnly<{
     mute: MuteState,
     subscriptions: Subscription[],
     ownUser: User,
+    unreadStreams: UnreadStreamsState,
     ...
   }>,
-  stream: string,
+  streamId: number,
   topic: string,
-|}): Button<HeaderArgs>[] => {
+|}): Button<TopicArgs>[] => {
+  const sub = subscriptions.find(x => x.stream_id === streamId);
+  invariant(sub, 'Subscription with provided stream id does not exist.');
```

Comment:
Just highlighting, that I have used an invariant here as well (this wasn't asked in @chrisbobbe's review). necessary for the subsequent call to `isTopicMuted`.
for the same reason as https://github.com/zulip/zulip-mobile/pull/4635#discussion_r632862010

Reply:
Thanks, writing this expectation explicitly like this is helpful. I think the expectation is not correct. :grin: I'll elaborate in a review comment.